static/js/TextsPropertiesMenu.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import ToggleSwitchLine from "./components/ToggleSwitchLine";
+import LayoutButtonLine from './LayoutButtonLine'
+import FontSizeButton from "./FontSizeButton";
+import {getLayoutOptions} from './constants';
+import PopoverMenu from "./components/PopoverMenu";
+import SourceTranslationsButtons from "./SourceTranslationsButtons";
+
+const calculateLayoutState = (showSource, sourceDir, showTranslation, translationDir) => {
+    return (!showSource || !showTranslation) ? 'mono' //one text
+        : (sourceDir !== translationDir) ? 'mixed' //two texts with different directions
+            : (sourceDir === 'rtl') ? 'bi-rtl' //two rtl texts
+                : 'bi-ltr'; //two ltr texts
+}
+
+function TextsPropertiesMenu({
+        sourceDir,
+        translationDir,
+        initialShowSource = true,
+        initialShowTranslation = false,
+        initialLayoutPreferences = {},
+        hasAliyot = false,
+        initialShowAliyot = true,
+        initialFontSize = 12,
+        hasVowels = false,
+        initialShowVowels = false,
+        hasCantilation = false,
+        initialShowCantilation = true,
+        hasPunctuation = false,
+        initialShowPunctuation = true
+    }) {
+    const [showSource, setShowSource] = useState(initialShowSource);
+    const [showTranslation, setShowTranslation] = useState(initialShowTranslation);
+    const setShowTexts = (source, translation) => {
+        setShowSource(source);
+        setShowTranslation(translation);
+    }
+
+    const layoutOptions = getLayoutOptions(sourceDir);
+    const [layoutsState, setLayoutsState] = useState(calculateLayoutState(showSource, sourceDir, showTranslation, translationDir));
+    useEffect(() => {
+      setLayoutsState(calculateLayoutState(showSource, sourceDir, showTranslation, translationDir));
+    }, [showSource, showTranslation]);
+
+    const [layoutPreferences, setLayoutPreferences] = useState(initialLayoutPreferences)
+    useEffect(() => {
+        setLayoutPreferences({
+            'mono': layoutPreferences.mono || 'continued',
+            'bi': layoutPreferences.bi || 'vertical',
+            'mixed': layoutPreferences.mixed
+        });
+    }, []);
+    const [layout, setLayout] = useState('');
+    useEffect(() => {
+        setLayout(layoutPreferences[layoutsState.replace(/-(?:ltr|rtl)/, '')] || layoutOptions[layoutsState][0]);
+    }, [layoutPreferences, layoutsState]);
+    const onLayoutChange = (newLayout) => {
+        setLayoutPreferences(prevState => ({
+            ...prevState,
+            [layoutsState]: newLayout
+        }))}
+
+    const [fontSize, setFontSize] = useState(initialFontSize);
+    const [showAliyot, setShowAliyot] = useState(initialShowAliyot);
+    const [showVowels, setShowVowels] = useState(initialShowVowels)
```

Comment:
see my comment on naming conventions with states. We usually add "state" at the end instead of "initial" as "initial" sounds like the data will change